pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -103,23 +109,34 @@ func (s *volumeStatCalculator) calcAndStoreStats() {
 			}
 			continue
 		}
-		stats = append(stats, s.parsePodVolumeStats(name, metric))
+		// Lookup the volume spec and add a 'PVCReference' for volumes that reference a PVC
+		volSpec := volumesSpec[name]
+		if pvcSource := volSpec.PersistentVolumeClaim; pvcSource != nil {
+			pvcRef := stats.PVCReference{
+				Name:      pvcSource.ClaimName,
+				Namespace: s.pod.GetNamespace(),
+			}
+			fsStats = append(fsStats, s.parsePodVolumeStats(name, &pvcRef, metric))
+		} else {
+			fsStats = append(fsStats, s.parsePodVolumeStats(name, nil, metric))
+		}
 	}
 
 	// Store the new stats
-	s.latest.Store(PodVolumeStats{Volumes: stats})
+	s.latest.Store(PodVolumeStats{Volumes: fsStats})
 }
 
 // parsePodVolumeStats converts (internal) volume.Metrics to (external) stats.VolumeStats structures
-func (s *volumeStatCalculator) parsePodVolumeStats(podName string, metric *volume.Metrics) stats.VolumeStats {
+func (s *volumeStatCalculator) parsePodVolumeStats(podName string, pvcRef *stats.PVCReference, metric *volume.Metrics) stats.VolumeStats {
 	available := uint64(metric.Available.Value())
 	capacity := uint64(metric.Capacity.Value())
 	used := uint64(metric.Used.Value())
 	inodes := uint64(metric.Inodes.Value())
 	inodesFree := uint64(metric.InodesFree.Value())
 	inodesUsed := uint64(metric.InodesUsed.Value())
 	return stats.VolumeStats{
-		Name: podName,
+		Name:   podName,
+		PVCRef: pvcRef,
```

Comment:
I am thinking to check pvcRef is nil or not here. If it is nil, return stats.VolumeStats without PVCRef field

Reply:
the result will be the same since PVCRef will default to nil whether we explicitly set `PVCRef: nil,` or not at all

Reply:
Is this needed - the behavior would be the same i.e VolumeStats.PVCRef will be nil in both cases?

Reply:
ok, I am hoping if PVCRef is nil, the printed result of volumestat will not show this field, instead of showing PVCRef: nil?

Reply:
👍 - it won't. It's tagged optional/omit-empty.